core/src/main/scala/akka/kafka/KafkaConsumerActor.scala

```diff
@@ -77,6 +80,8 @@ private[kafka] class KafkaConsumerActor[K, V](settings: ConsumerSettings[K, V])
   var requests = Map.empty[ActorRef, RequestMessages]
   var consumer: KafkaConsumer[K, V] = _
   var commitsInProgress = 0
+  val MaxWakeups = 3
```

Comment:
I left this here to implement later as configurable, if the core algorithm passes review. Will make it configurable.